driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -58,8 +59,7 @@
     private final io.netty.util.Timer scheduler;
 
     private volatile List<Host> triedHosts;
-
-    private volatile Map<InetSocketAddress, Throwable> errors;
```

Comment:
Good point, but the ideal capacity / concurrency is related to the query plan size and the speculative execution policy... difficult to guess sensible values here. Maybe use an atomic reference? or a volatile + double checked locking?